internal/codespaces/api/api.go

```diff
@@ -60,10 +67,24 @@ type httpClient interface {
 
 // New creates a new API client with the given token and HTTP client.
 func New(token string, httpClient httpClient) *API {
+	serverURL := os.Getenv("GITHUB_SERVER_URL")
```

Comment:
Just a thought, would it be better to pass these in and defer reading from env to the caller? It could make our tests easier if we wanted to use something httptest or something, i know we can set env vars there as well but curious what you think.

Reply:
Agreed that it would be best if the API client was agnostic to the environment configuration and instead just accepted a configuration from the caller. In general, lower-level modules should not ever read from the environment directly.

Perhaps the `api.API` struct can have fields like `ServerURL` that can be overridden after calling `api.New()`?

Reply:
@mislav @josebalius ok i pulled out the os.Getenv into the caller. Let me know if that looks good. 

> Perhaps the api.API struct can have fields like ServerURL that can be overridden after calling api.New()?

I typically think it's better that you either create a constructor or have exposed fields. This way it's clear that you only have one way to create the targeted type.